cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/package.scala

```diff
@@ -51,15 +50,9 @@ package object kubernetes extends LazyLogging {
   private val EndpointTrimChars = Set('_', '-')
 
   /**
-   * This is the main method which generates the kubernetes resources.
-   *
-   * This method accepts a docker config supplied by `getDockerConfig`, generates the Kubernetes resources, and
-   * supplies these generated resources to the `outputHandler`.
-   *
-   * The `getDockerConfig` and `outputHandler` are supplied as function, and when `generateDeploymentArgs` and
-   * `kubernetesArgs` are supplied, the function is invoked and the resulting [[Try[Done]] is returned.
+   * This is the main method which generates the Kubernetes resources.
    */
-  def generateResources(getDockerConfig: String => Try[Config], outputHandler: Seq[GeneratedKubernetesResource] => Unit)(generateDeploymentArgs: GenerateDeploymentArgs, kubernetesArgs: KubernetesArgs): Try[Done] = {
+  def generateResources(config: Config, generateDeploymentArgs: GenerateDeploymentArgs, kubernetesArgs: KubernetesArgs): Try[Seq[GeneratedKubernetesResource]] = {
```

Comment:
Just accept the config directly, I don't think we need to couple tests so much to ensure that the program is passing the correct argument. Also, just return the resources directly and let client code determine what to do with it, instead of having fairly limiting higher-order functions.

Basically, I'm asserting plain boring functions are more testable and readable, hence the change here.